CAIPs/caip-25.md

```diff
@@ -16,422 +16,247 @@ CAIP-25 defines an authorization procedure for a chain agnostic provider to inte
+- Update an existing session by including the same `sessionId`
+- Create a new session if no `sessionId` is provided (discouraged)
 
-When a caller wishes revoke an active session, it can do so by calling [`wallet_revokeSession`][CAIP-285].
-- When a `sessionId` is returned in the initial `wallet_createSession` response, the caller MUST call `wallet_revokeSession` with the supplied `sessionId` to revoke that session.
-- When the wallet does not provide a `sessionId` in its initial response, a call to `wallet_revokeSession` revokes the single active session between caller and wallet.
+Session updates triggered by the wallet must notify the caller using `wallet_sessionChanged`.
 
-For more detail on the lifecycle and management of sessions with and without `sessionId`s, see the informational [CAIP-316][].
+Callers may revoke sessions using `wallet_revokeSession`, either by `sessionId` or directly if no sessionId was established.
 
 #### Session Data and Metadata
 
-Initial and ongoing authorization requests are grouped into two top-level objects containing keyed [scopeObjects][CAIP-217], named `requiredScopes` and `optionalScopes`
-respectively.
-Each `scopeObject` in either parent object MUST be keyed uniquely within its parent, but these keys CAN appear in both
-(i.e., additional properties of an authorization target in `requiredScopes` may be requested in a separate `scopeObject` with the same key in the `optionalScopes` array).
+Authorization requests are expressed as a top-level object `scopes` containing keyed [scopeObjects][CAIP-217].
 
-Each `scopeObject` in these parent `...Scopes` objects can be keyed to a specific [CAIP-2][] network identifier, or to an entire [CAIP-104][] namespace.
-`scopeObjects` keyed to an entire [CAIP-104][] namespace SHOULD contain a non-empty `references` array to be actionable, making them functionally equivalent to a series of identical `scopeObjects`, each keyed to one of the members of `references` expressed as a [CAIP-2][] scope.
-An empty or absent `references` array SHOULD NOT be interpreted as a namespace-wide authorization (i.e. authorization for ANY network therein), but rather as a null authorization of 0 specified `references`s within that namespace.
-(See [CAIP-217][] for more details on the structure of the typed objects included in these `...Scopes` objects.)
+Each `scopeObject` is keyed by a [CAIP-2][] chain ID.
 
-The distinction between `requiredScopes` and `optionalScopes` is ultimately semantic, since a wallet may still choose to establish a connection authorizing a subset of requested networks or requested capabilities from each; the primary function of the distinction is to offer callers a mechanism for signaling which scopes they consider primary and which they consider secondary to their request, in order to better inform the authorization logic of the respondent.
+Wallets MAY authorize a subset of scopes. This enables granular control and flexibility.
```

Comment:
We allow subset, superset, or entirely disjoint set to be granted. The dapp requested scopes are only taken as a suggestion on what defaults to show to the user. The user is free to deviate from the default as they wish.

Reply:
This wording works for me